stream.go

```diff
@@ -202,11 +203,14 @@ func (st *Stream) produceKVs(ctx context.Context, threadId int) error {
 
 		// This unique stream id is used to identify all the keys from this iteration.
 		streamId := atomic.AddUint32(&st.nextStreamId, 1)
+		var scanned int
 
 		sendIt := func() error {
 			select {
 			case st.kvChan <- outList:
 				outList = z.NewBuffer(2 * batchSize)
+				atomic.AddUint64(&st.scanned, uint64(itr.scanned-scanned))
+				scanned = itr.scanned
```

Comment:
updating `st.scanned` at `sendIt` instead of iteration loop below because we need not frequently update it as it is just an estimation.